src/foam/core/Enum.js

```diff
@@ -205,20 +205,26 @@ foam.CLASS({
         for ( var i = 0 ; i < v.length ; i++ ) {
           var def = v[i];
 
+          if ( foam.Object.isInstance(def) && def.class ) {
```

Comment:
You should be able to merge this case and the EnumValueAxim.isInstance case to a single ternary like we do in FObjectProperty.adapt.